pkg/antctl/raw/multicluster/join.go

```diff
@@ -0,0 +1,318 @@
+		return fmt.Errorf("the member Namespace cannot be empty")
+	}
+
+	return nil
+}
+
+func yamlUnmarshall(raw []byte, v interface{}) error {
+	d := yaml.NewDecoder(bytes.NewReader(raw))
+	d.KnownFields(false)
+	return d.Decode(v)
+}
+
+var joinExamples = strings.Trim(`
+# Join the ClusterSet from a member cluster with command line options with secret.
+  $ antctl mc join --leader-namespace=antrea-multicluster \
+				   --leader-clusterid=test-cluster-north \
+				   --secret=member-access-token \
+				   --leader-server=https://172.18.0.3:6443 \
+				   --clusterset=test-clusterset \
+				   --namespace=kube-system \
+				   --clusterid=test-cluster-east
+
+# Join the ClusterSet from a member cluster with command line options with token.
+  $ antctl mc join --leader-namespace=antrea-multicluster \
+				   --leader-clusterid=test-cluster-north \
+				   --leader-server=https://172.18.0.3:6443 \
+				   --clusterset=test-clusterset \
+				   --namespace=kube-system \
+				   --clusterid=test-cluster-east \
+                   --secret-file=member-access-token.yaml
+
+# Join the ClusterSet from a member cluster with a config file. If both command line options and config file are specified, 
+the arguments in the config file will be used.
+  $ antctl mc join --config-file ./config.yaml
+
+# Config file example:
+apiVersion: multicluster.antrea.io/v1alpha1
+kind: ClusterSetConfig
+memberCluster:
+  namespace: kube-system
+  clusterID: test-cluster-east
+leaderCluster:
+  namespace: antrea-multicluster
+  secret: member-access-token
+  apiServer: https://172.18.0.3:6443
+  clusterID: test-cluster-north
+clusterSetID: test-clusterset
+`, "\n")
+
+func NewJoinCommand() *cobra.Command {
+	command := &cobra.Command{
+		Use:     "join",
+		Short:   "Join the ClusterSet from a member cluster",
+		Args:    cobra.MaximumNArgs(0),
+		Example: joinExamples,
+		RunE:    joinRunE,
+	}
+
+	o := JoinOptions{}
+	joinOpts = &o
+	command.Flags().StringVarP(&joinOpts.LeaderClusterOpt.Namespace, "leader-namespace", "", "", "Namespace of the leader cluster")
+	command.Flags().StringVarP(&joinOpts.LeaderClusterOpt.ClusterID, "leader-clusterid", "", "", "Cluster ID of the leader cluster")
+	command.Flags().StringVarP(&joinOpts.LeaderClusterOpt.Secret, "secret", "", "", "Secret of the leader cluster")
+	command.Flags().StringVarP(&joinOpts.LeaderClusterOpt.APIServer, "leader-server", "", "", "API Server endpoint of the leader cluster")
+	command.Flags().StringVarP(&joinOpts.MemberClusterOpt.Namespace, "namespace", "n", "", "Namespace of the member cluster")
+	command.Flags().StringVarP(&joinOpts.MemberClusterOpt.ClusterID, "clusterid", "", "", "Cluster ID of the member cluster")
+	command.Flags().StringVarP(&joinOpts.ClusterSetID, "clusterset", "", "", "ClusterSet ID")
+	command.Flags().StringVarP(&joinOpts.SecretFile, "secret-file", "", "", "Secret file to leader cluster. If both Secret and Secret file are specified, the Secret file will be used.")
```

Comment:
Name it "token-secret-file"

"YAML file defines the leader token Secret resource. If both Secret name and Secret file are..."